satellite/notification/service.go

```diff
@@ -0,0 +1,134 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package notification
+
+import (
+	"context"
+	"net"
+	"sync"
+	"time"
+
+	"github.com/zeebo/errs"
+	"go.uber.org/zap"
+	"gopkg.in/spacemonkeygo/monkit.v2"
+
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/rpc"
+	"storj.io/storj/private/sync2"
+	"storj.io/storj/satellite/mailservice"
+	"storj.io/storj/satellite/overlay"
+)
+
+var (
+	// Error is the default error class for notification package.
+	Error = errs.Class("notification")
+
+	mon = monkit.Package()
+)
+
+// Service is the notification service between storage nodes and satellites.
+// architecture: Service
+type Service struct {
+	log     *zap.Logger
+	dialer  rpc.Dialer
+	overlay *overlay.Service
+	mailer  *mailservice.Service
+
+	loop *sync2.Cycle
+	lock *sync.Mutex
+}
+
+// NewService creates a new notification service.
+func NewService(log *zap.Logger, dialer rpc.Dialer, overlay *overlay.Service, mail *mailservice.Service) *Service {
+	return &Service{
+		log:     log,
+		dialer:  dialer,
+		overlay: overlay,
+		mailer:  mail,
+		lock:    &sync.Mutex{},
+	}
+}
+
+// Run sets the Rate Limiter up to ensure we dont spam.
+func (service *Service) Run(ctx context.Context) (err error) {
+	service.log.Debug("Starting Rate Limiter")
+	service.loop = sync2.NewCycle(1 * time.Hour)
+
+	err = service.loop.Run(ctx, nil)
+	return err
+}
+
+// Close closes the resources.
+func (service *Service) Close() error {
+	service.loop.Stop()
+	service.loop.Close()
+
+	return nil
+}
```

Comment:
I dont think we are using the entire loop at this stage, but i guess its fine to leave it in for now